include/SinglePDF.h

```diff
@@ -81,7 +81,11 @@ class UniformPDF: public SinglePDF {
 class VectorPDF: public TObject {
  public:
   VectorPDF() {};
-  ~VectorPDF() {};
+  ~VectorPDF() {
+    for(auto pdf : m_Members) {
+      delete pdf;
+    }
+  };
```

Comment:
Since this class inherits from `TObject`, does streaming instances of `VectorPDF` to ROOT files still work? If not, or you don't want to test, perhaps use `DISABLE_ROOT_IO`:
```suggestion
  ~VectorPDF() {
#ifdef DISABLE_ROOT_IO
    for(auto pdf : m_Members) {
      delete pdf;
    }
#endif
  };
```